lib/ansible/constants.py

```diff
@@ -333,6 +333,7 @@ def load_config_file():
 # that it may be a security risk to do so.
 ANSIBLE_SSH_CONTROL_PATH       = get_config(p, 'ssh_connection', 'control_path', 'ANSIBLE_SSH_CONTROL_PATH', u"%(directory)s/ansible-ssh-%%h-%%p-%%r")
 ANSIBLE_SSH_CONTROL_PATH_DIR   = get_config(p, 'ssh_connection', 'control_path_dir', 'ANSIBLE_SSH_CONTROL_PATH_DIR', u'~/.ansible/cp')
+ANSIBLE_SSH_CONTROL_PATH_FIX   = get_config(p, 'ssh_connection', 'control_path_fix', 'ANSIBLE_SSH_CONTROL_PATH_FIX', False, value_type='boolean')
```

Comment:
Rather than set a fairly esoteric new config item, why not just change default control path to either be much shorter or use %C.

I just don't think this fix is that discoverable, and once you've discovered it, you may as well set the other values instead.